### ISSUE TITLE
Improve build and instructions

### ISSUE THREAD

**Issue by fredRos** [eos/eos#46]
On Ubuntu 16.04, `./configure` runs without complaints but during compilation it fails because `libyaml-cpp-dev` isn't installed. This should be checked at configure time and mentioned as a requirement in the manual
```
parameters.cc:33:27: fatal error: yaml-cpp/yaml.h: No such file or directory
```

**Comment by fredRos** [comment]
A warning came up after I installed `yaml-cpp`
```
hard-scattering.cc: In static member function ‘static std::complex<double> eos::HardScattering::j3_massless(const double&, const double&, const double&, const double&, const double&)’:
hard-scattering.cc:278:38: warning: unused variable ‘ubar2’ [-Wunused-variable]
         const double ubar = 1.0 - u, ubar2 = ubar * ubar;
```
Should be easy to fix. A copy-paste error?

**Comment by fredRos** [comment]
The manual instructions to use packagecloud aren't sufficient because the repo is not verifiable. We should instead refer to the instructions here https://packagecloud.io/eos/eos/install

**Comment by fredRos** [comment]
With the package cloud, the build works but I get a warning from libtool

     libtool: warning: library '/usr/lib/libMinuit2.la' was moved.

Is that something to worry about?

**Comment by dvandyk** [comment]
Let's go through it piece by piece:

1. There is already a configure check for ```yaml-cpp``` using ```AC_CHECK_LIB```. I added a further check for the header in 62fcc96.
2. The unused variable in ```eos/rare-b-decays/hard-scattering.cc``` has been removed in b302736.
3. I think the packagecloud instructions are insance to download a bash script and run it with superuser permissions, without the user being able to see what programms are run. I added the relevant call to ```apt-key``` to the manual's installation instructions, see ea3593e.
4. No need to worry about the libtool warning. I still have to fix how we build Minuit2 though for distribution. Will keep track of that in a separate issue though.
